diff --git a/node_modules/@graphql-mesh/odata/index.js b/node_modules/@graphql-mesh/odata/index.js
index 7132bd9..71f0be6 100644
--- a/node_modules/@graphql-mesh/odata/index.js
+++ b/node_modules/@graphql-mesh/odata/index.js
@@ -135,7 +135,7 @@ class ODataHandler {
                     : this.config.customFetch;
         }
         else {
-            fetch = utils.getCachedFetch(this.cache);
+            fetch = utils.getCachedFetch(this.cache, this.config.rejectUnauthorized);
         }
         const { baseUrl: nonInterpolatedBaseUrl, operationHeaders } = this.config;
         const baseUrl = utils.stringInterpolator.parse(nonInterpolatedBaseUrl, {
diff --git a/node_modules/@graphql-mesh/odata/index.mjs b/node_modules/@graphql-mesh/odata/index.mjs
index 33c6d3e..68a43c6 100644
--- a/node_modules/@graphql-mesh/odata/index.mjs
+++ b/node_modules/@graphql-mesh/odata/index.mjs
@@ -131,7 +131,7 @@ class ODataHandler {
                     : this.config.customFetch;
         }
         else {
-            fetch = getCachedFetch(this.cache);
+            fetch = utils.getCachedFetch(this.cache, this.config.rejectUnauthorized);
         }
         const { baseUrl: nonInterpolatedBaseUrl, operationHeaders } = this.config;
         const baseUrl = stringInterpolator.parse(nonInterpolatedBaseUrl, {
